client/block-builder/src/lib.rs

```diff
@@ -135,6 +135,8 @@ pub struct BlockBuilder<'a, Block: BlockT, A: ProvideRuntimeApi<Block>, B> {
 	block_id: BlockId<Block>,
 	parent_hash: Block::Hash,
 	backend: &'a B,
+	/// The estimated size of the block header.
+	estimated_header_size: usize,
```

Comment:
I am OK with this but it would probably be fine if we just computed it anyway anytime it is required in `estimate_block_size`.

Reply:
PERFORMANCE OPTIMIZATIONS!

(And we also don't store the header)

Reply:
Ah yeah, I missed that part 😅